### PR TITLE
Homepage: Display bhm link instead of video for en-US

### PR DESCRIPTION
**Pull request by maureensturgeon**
For en-US locale, display Celebrate Black History Month with us! link which links to `/blackhistorymonth` on the homepage.

This was tested locally (couldn't find homepage tests to add to).

Screen Shot 2021-02-15 at 9 35 52 AM

Screen Shot 2021-02-15 at 9 35 45 AM

Screen Shot 2021-02-15 at 9 35 13 AM

 

## Links

- [spec]()
- [jira](https://codedotorg.atlassian.net/browse/LP-1747)

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: pegasus/sites.v3/code.org/views/homepage_hero.haml [hunk 1]
```diff
@@ -97,6 +97,10 @@
         %div
           %a{onclick: "return showVideo_mainvideo();", style: "cursor: pointer"}
             %img{src: "images/homepage/watch_video_drawn_button.png", style: "width: 250px; margin-top: 10px"}
+      - elsif entry[:type] == "celebrate_bhm"
+        %a.herohocbutton.bhmbutton{href: "/blackhistorymonth"}
+          %i.fa.fa-arrow-circle-o-right
+          %span Celebrate black history month with us!
```
Comment (jamescodeorg) [inline pegasus/sites.v3/code.org/views/homepage_hero.haml]: Should "Black History Month" be capitalized?

File: pegasus/sites.v3/code.org/views/homepage_hero.haml [hunk 2]
```diff
@@ -97,6 +97,10 @@
         %div
           %a{onclick: "return showVideo_mainvideo();", style: "cursor: pointer"}
             %img{src: "images/homepage/watch_video_drawn_button.png", style: "width: 250px; margin-top: 10px"}
+      - elsif entry[:type] == "celebrate_bhm"
+        %a.herohocbutton.bhmbutton{href: "/blackhistorymonth"}
```
Comment (jamescodeorg) [inline pegasus/sites.v3/code.org/views/homepage_hero.haml]: Visually, something looks a bit off?  The icon and text don't look quite aligned vertically and the icon might be too "bold"?  (Sorry for the nitpick, I used to be a newspaper layout editor.)  At first, I thought it was the font that was too small but I see that it's the same as the font that's currently being used for "Watch the video".  Maybe there's something we can adjust with the new icon?

File: pegasus/src/homepage.rb [hunk 3]
```diff
@@ -184,21 +184,28 @@ def self.get_actions(request)
...
+        download_path: download_path,
+        facebook: facebook,
+        twitter: twitter
+      }
+
+      bhm_action = {
+        type: "celebrate_bhm",
+      }
+
+      actions.push(I18n.locale == :"en-US" ? bhm_action : code_video_action)
```
Comment (jamescodeorg) [inline pegasus/src/homepage.rb:211]: (very minor) Is locale a symbol or a string?  (In most places, I see us just compare to `en-US` but don't know which is more correct.  Either way, not too important.)

File: pegasus/src/homepage.rb [hunk 4]
```diff
@@ -184,21 +184,28 @@ def self.get_actions(request)
...
+        text: "homepage_action_text_codevideo",
+        type: "video",
+        youtube_id: youtube_id,
+        download_path: download_path,
+        facebook: facebook,
+        twitter: twitter
+      }
+
+      bhm_action = {
+        type: "celebrate_bhm",
```
Comment (jamescodeorg) [inline pegasus/src/homepage.rb]: (minor) It seems a little weird to me that this `type` is so specific when the other types are a bit more generic, but up to you.

File: pegasus/sites.v3/code.org/views/homepage_hero.haml (hunk 1 above)
Comment (maureensturgeon) [reply pegasus/sites.v3/code.org/views/homepage_hero.haml]: That was an oversight, good catch!

File: pegasus/src/homepage.rb (hunk 3 above)
Comment (maureensturgeon) [reply pegasus/src/homepage.rb:211]: Symbol, I tried as a string, it didn't work

File: pegasus/src/homepage.rb (hunk 4 above)
Comment (maureensturgeon) [reply pegasus/src/homepage.rb]: I see what you mean, I could make it more generic.

File: pegasus/src/homepage.rb (hunk 3 above)
Comment (bencodeorg) [reply pegasus/src/homepage.rb:211]: Also curious whether there's a standard of how we determine locale? I see elsewhere in that same file we look at request.locale, do you happen to know the difference?

**Comment by bencodeorg** [comment]
One general question...did you get a chance to see how this looks on smaller screens?

**Comment by maureensturgeon** [comment]
> One general question...did you get a chance to see how this looks on smaller screens?

I've added screenshots of other sizes to the description

**Review by bencodeorg — APPROVED** [review]
LGTM! Can't tell if the vertical spacing issue is totally fixed from screenshots or if they're just old, but don't think that should block it anyway.

File: pegasus/src/homepage.rb (hunk 3 above)
Comment (maureensturgeon) [reply pegasus/src/homepage.rb:211]: I do not - I'm assuming that the i18n locale is somehow set by the request locale?

Comment (bencodeorg) [reply pegasus/src/homepage.rb:211]: Sounds plausible :)

File: pegasus/sites.v3/code.org/views/homepage_hero.haml (hunk 2 above)
Comment (maureensturgeon) [reply pegasus/sites.v3/code.org/views/homepage_hero.haml]: You are correct! I was missing a vertical-align: middle on that span of text, should be fixed now! Good eye

File: pegasus/sites.v3/code.org/public/css/homepage.css [hunk 5]
```diff
@@ -355,6 +355,25 @@ body.modal-open .supreme-container{
...
+  font-size: 18px;
+  color: white;
+  display: inline-block;
+  margin-top: 24px
+}
+
+#homepage #actions .herolink-white i {
+  vertical-align: middle;
+  margin-right: 5px;
+  font-size: 26px;
```
Comment (jamescodeorg) [inline pegasus/sites.v3/code.org/public/css/homepage.css:370]: Just curious, was 26px just the size that made things look right or was there a reason for picking this value?

**Review by jamescodeorg — APPROVED** [review]
LGTM

File: pegasus/sites.v3/code.org/public/css/homepage.css (hunk 5 above)
Comment (maureensturgeon) [reply pegasus/sites.v3/code.org/public/css/homepage.css:370]: This is specifically for the icon style `i`, I just looked at the mocks in the ticket and it was the size that looked most like the icon in the mocks.

**Pull request merged**